contracts/integrations/custom/CustomIntegrationLiquity.sol

```diff
@@ -0,0 +1,232 @@
+        liquityBorrowerOperations = _liquityBorrowerOperations;
+    }
+
+    /* =============== Internal Functions ============== */
+
+    /**
+     * Whether or not the data provided is valid
+     *
+     * hparam  _data                     Data provided
+     * @return bool                      True if the data is correct
+     */
+    function _isValid(bytes memory _data) internal view override returns (bool) {
+        (, uint amountLUSDToBorrow) = abi.decode(_data, (address, uint));
+        return amountLUSDToBorrow >= liquityBorrowerOperations.MIN_NET_DEBT();
+    }
+
+    /**
+     * Which address needs to be approved (IERC-20) for the input tokens.
+     *
+     * hparam  _data                     Data provided
+     * hparam  _opType                   O for enter, 1 for exit
+     * @return address                   Address to approve the tokens to
+     */
+    function _getSpender(
+        bytes calldata, /* _data */
+        uint8 /* _opType */
+    ) internal view override returns (address) {
+        // No approval needed, as we are lending ETH
+        // Can't use address(0), so just use the borrower operations
+        return address(liquityBorrowerOperations);
+    }
+
+    /**
+     * The address of the IERC-20 token obtained after entering this operation
+     *
+     * hparam  _token                     Address provided as param
+     * @return address                    Address of the resulting lp token
+     */
+    function _getResultToken(address) internal view override returns (address) {
+        return liquityBorrowerOperations.lusdToken();
+    }
+
+    /**
+     * Return enter custom calldata
+     *
+     * hparam  _strategy                 Address of the strategy
+     * hparam  _data                     OpData e.g. Address of the pool
+     * hparam  _resultTokensOut          Amount of result tokens to send
+     * hparam  _tokensIn                 Addresses of tokens to send to spender to enter
+     * hparam  _maxAmountsIn             Amounts of tokens to send to spender
+     *
+     * @return address                   Target contract address
+     * @return uint256                   Call value
+     * @return bytes                     Trade calldata
+     */
+    function _getEnterCalldata(
+        address, /* _strategy */
+        bytes calldata _data,
+        uint256, /* _resultTokensOut */
+        address[] calldata, /* _tokensIn */
+        uint256[] calldata _maxAmountsIn
+    )
+        internal
+        view
+        override
+        returns (
+            address,
+            uint256,
+            bytes memory
+        )
+    {
+        (, uint amountLUSDToBorrow) = abi.decode(_data, (address, uint));
+
+        return (address(liquityBorrowerOperations), _maxAmountsIn[0], abi.encodeWithSelector(
+          IBorrowerOperations.openTrove.selector,
+          LIQUITY_MAX_FEE_PERCENTAGE, // maxFeePercentage
+          amountLUSDToBorrow, // LUSD amount
+          address(0), // lowerHint
```

Comment:
It would be more gas efficient to calculate these somewhere in our UI, but I assume that is off the table for now.

ref. https://github.com/liquity/beta/blob/main/README.md#supplying-hints-to-trove-operations

Reply:
yes, we need to calculate it here